mlir/lib/Dialect/MIOpen/Transforms/AlignTiling.cpp

```diff
@@ -200,6 +202,136 @@ template <typename T> struct MILARewritePattern : public OpRewritePattern<T> {
     return inp;
   }
 
+  Value createCollapseShapeOp(PatternRewriter &b, Location loc,
+                              Value source) const {
+    auto ctx = b.getContext();
+    auto sourceType = source.getType().cast<ShapedType>();
+    assert(sourceType.hasStaticShape() &&
+           "Only memrefs with static shapes are allowed");
+
+    auto shape = sourceType.getShape();
+    uint64_t collapsedDim = 1;
+    SmallVector<AffineExpr, 2> exprs;
+    for (uint32_t dim = 0; dim < shape.size(); ++dim) {
+      collapsedDim *= shape[dim];
+      exprs.push_back(getAffineDimExpr(dim, ctx));
+    }
+
+    SmallVector<int64_t, 1> collapsedShape;
+    SmallVector<ReassociationExprs, 1> reassocs;
+    collapsedShape.push_back(collapsedDim);
+    reassocs.push_back(exprs);
+
+    auto collapsedType =
+        MemRefType::get(collapsedShape, sourceType.getElementType(), {}, 5);
+    Value result =
+        b.create<memref::CollapseShapeOp>(loc, collapsedType, source, reassocs);
```

Comment:
Are we not doing this with transforms? Or is that a future PR?

Reply:
Not doing it with transform because it's only consumed by linalg.